contracts/Credential.sol

```diff
@@ -16,19 +18,43 @@ contract Credential {
     string courseName;
     string degreeLevel;
     string endorserName;
+    string issuerName; //name of Institution
     uint issuanceDate;
     uint expiryDate;
     credentialState state;
     address issuer; //address of Institution
-    address owner;
+    address owner; //student (recipient of credential)
   }
 
+  event add_credential(
+    uint256 credId,
+    address issuer,
+    string issuerName,
+    address owner,
+    string studentName,
+    string courseName
+  );
+  event delete_credential(
+    address issuer,
+    string issuerName,
+    address owner,
+    string studentName,
+    string courseName
+  );
+  event revoke_credential(
+    address issuer,
+    string issuerName,
+    address owner,
+    string studentName,
+    string courseName
+  );
+
   uint256 public numCredentials = 0;
   mapping(uint256 => credential) public credentials;
 
   Institution institutionContract;
 
-  constructor(Institution insitutionAddr) public {
+  constructor(Institution insitutionAddr) {
```

Comment:
removed `public` visibility?

Reply:
removed it because of a compiler warning but added it back